src/channel.h

```diff
@@ -76,7 +76,7 @@ class CChannel : public QObject
 
     void PrepAndSendPacket ( CHighPrioSocket* pSocket, const CVector<uint8_t>& vecbyNPacket, const int iNPacketLen );
 
-    void ResetTimeOutCounter() { iConTimeOut = iConTimeOutStartVal; }
+    void ResetTimeOutCounter() { iConTimeOut = bDisconnectAndDisable ? 1 : iConTimeOutStartVal; }
```

Comment:
I don't like this. Should probably be expanded into an assignment and boolean comparison.

Reply:
Yeah, I not enthused by the whole `bDisconnectAndDisable` -- I can't see the problem it's trying to fix.

Reply:
I think they wanted to introduce an intermediate state somehow to fix the button being able to get out of sync.
From https://github.com/jamulussoftware/jamulus/pull/2550#issue-1181299521

> Added bDisconnectAndDisable to CChannel. (For a Client now
Channel.Disconnect() will block audio data and auto disable the channel on disconnected)